twine-upload.sh

```diff
@@ -40,6 +40,13 @@ INPUT_VERIFY_METADATA="$(get-normalized-input 'verify-metadata')"
 INPUT_SKIP_EXISTING="$(get-normalized-input 'skip-existing')"
 INPUT_PRINT_HASH="$(get-normalized-input 'print-hash')"
 
+PASSWORD_DEPRECATION_NUDGE="::warning title=Password-based uploads deprecated::\
+Starting in 2024, PyPI will require all users to enable Two-Factor \
+Authentication. This will consequently require all users to switch \
+to either API tokens or Trusted Publishers (preferred) for package \
```

Comment:
Since this is GitHub Actions we're warning in, let's put Trusted Publishers only, similar to the "Trusted Publishers" nudge? (Do we have a link to API keys for folks using reusable workflows in the TP docs?)

Also, is this message worthy of an `::error`? Does that message type fail a run or only make the message red?

Reply:
> Since this is GitHub Actions we're warning in, let's put Trusted Publishers only, similar to the "Trusted Publishers" nudge? (Do we have a link to API keys for folks using reusable workflows in the TP docs?)

Yeah, this was my only lingering concern -- by nudging TP only, we might confuse some users of reusable workflows who can't switch yet. 

The reusable workflow limitation *is* documented, but not super prominently: <https://docs.pypi.org/trusted-publishers/troubleshooting/#reusable-workflows-on-github>

So TL;DR no strong opinion from me -- if you'd rather not mention API tokens here that SGTM, but we should probably be prepared for a ~few users being confused 🙂 

> Also, is this message worthy of an ::error? Does that message type fail a run or only make the message red?

I believe it just makes the message red, so `::error` works for me!

Reply:
Whoops, this completely fell off my stack. Making these suggested changes now!